pkgs/servers/mlflow-server/default.nix

```
@@ -1,36 +1,41 @@
-{ python3, writeText}:
+{
+  python3Packages,
+  writeText,
+}:
 
 let
-  py = python3.pkgs;
+  py = python3Packages;
 in
-py.toPythonApplication
-  (py.mlflow.overridePythonAttrs(old: rec {
+py.toPythonApplication (
+  py.mlflow.overridePythonAttrs (old: rec {
 
-    propagatedBuildInputs = old.propagatedBuildInputs ++ [
+    propagatedBuildInputs = old.dependencies ++ [
       py.boto3
       py.mysqlclient
     ];
 
-    postPatch = (old.postPatch or "") + ''
-      substituteInPlace mlflow/utils/process.py --replace \
-        "child = subprocess.Popen(cmd, env=cmd_env, cwd=cwd, universal_newlines=True," \
-        "cmd[0]='$out/bin/gunicornMlflow'; child = subprocess.Popen(cmd, env=cmd_env, cwd=cwd, universal_newlines=True,"
-    '';
-
-    gunicornScript = writeText "gunicornMlflow"
-    ''
-        #!/usr/bin/env python
-        import re
-        import sys
-        from gunicorn.app.wsgiapp import run
-        if __name__ == '__main__':
-          sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', ''', sys.argv[0])
-          sys.exit(run())
+    postPatch =
+      (old.postPatch or "")
+      + ''
+        substituteInPlace mlflow/utils/process.py --replace \
+          "child = subprocess.Popen(cmd, env=cmd_env, cwd=cwd, universal_newlines=True," \
+          "cmd[0]='$out/bin/gunicornMlflow'; child = subprocess.Popen(cmd, env=cmd_env, cwd=cwd, universal_newlines=True,"
       '';
 
+    gunicornScript = writeText "gunicornMlflow" ''
+      #!/usr/bin/env python
```

```suggestion
      #!${python.interpreter}
```

Why is that not in the entrypoint of the build system?

---

No idea... I applied your suggestion.

---

Can you try adding an entrypoint to setup.cfg or pyptoject.toml?
It would be really cool to avoid the python code hete